codec/codec_test.go

```diff
@@ -79,6 +81,63 @@ func TestMetrics(t *testing.T) {
 	}, metric, metricdatatest.IgnoreTimestamp())
 }
 
+func BenchmarkEncode(b *testing.B) {
+	ev := fullEvent(b)
+	testCases := map[string]struct {
+		codec Codec
+	}{
+		"json": {
+			codec: JSON{},
+		},
+		"vtproto": {
+			codec: VTProto{},
+		},
+	}
+	for name, tc := range testCases {
+		b.Run("format="+name, func(b *testing.B) {
+			var output atomic.Int64
+			b.RunParallel(func(p *testing.PB) {
+				var localOutput int64
+				for p.Next() {
+					by, _ := tc.codec.Encode(ev)
+					localOutput += int64(len(by))
+				}
+				output.Add(localOutput)
+			})
+			bytePerOp := float64(output.Load()) / float64(b.N)
+			b.ReportMetric(bytePerOp, "bytes/op")
+		})
+	}
+}
+
+func BenchmarkDecode(b *testing.B) {
+	ev := fullEvent(b)
+	testCases := map[string]struct {
```

Comment:
Same here, seems like using a struct with a single field vs the field directly may be more straightforward?